src/ResourceManager/RecoveryServices.SiteRecovery/Commands.RecoveryServices.SiteRecovery/ReplicationProtectedItem/UpdateAzureRmRecoveryServicesAsrProtectionDirection.cs

```diff
@@ -432,22 +508,159 @@ private void SetRPReprotect()
+                                    disk.VhdUri));
+                        }
+
+                        if (string.IsNullOrEmpty(disk.RecoveryAzureStorageAccountId))
+                        {
+                            throw new PSArgumentException(
+                                string.Format(
+                                    Properties.Resources.InvalidRecoveryAzureStorageAccountIdDiskInput,
+                                    disk.VhdUri));
+                        }
+
+                        a2aSwitchInput.VmDisks.Add(new A2AVmDiskInputDetails
+                        {
+                            DiskUri = disk.VhdUri,
+                            RecoveryAzureStorageAccountId = disk.RecoveryAzureStorageAccountId,
+                            PrimaryStagingAzureStorageAccountId =
+                                disk.LogStorageAccountId
+                        });
+                    }
+                }
+
+                input.Properties.ProviderSpecificDetails = a2aSwitchInput;
+            }
+
+            var response =
+                RecoveryServicesClient.StartSwitchProtection(
+                this.fabricName,
+                this.protectionContainerName,
+                input);
+
+            var jobResponse =
+                RecoveryServicesClient
+                .GetAzureSiteRecoveryJobDetails(
+                    PSRecoveryServicesClient.GetJobIdFromReponseLocation(response.Location));
+
+            WriteObject(new ASRJob(jobResponse));
+        }
+
+        /// <summary>
+        ///     validate replicated protected item parameter values depending on switch parameter.
+        /// </summary>
         private void validateRPISwitchParam()
         {
-            if ((this.AzureToVMware.IsPresent 
+            if ((this.AzureToVMware.IsPresent
                 && !this.ReplicationProtectedItem.ReplicationProvider.Equals(Constants.InMageAzureV2))
-                || (this.VMwareToAzure.IsPresent 
+                || (this.VMwareToAzure.IsPresent
                 && !this.ReplicationProtectedItem.ReplicationProvider.Equals(Constants.InMage))
-                || (this.HyperVToAzure.IsPresent 
+                || (this.HyperVToAzure.IsPresent
                 && !this.ReplicationProtectedItem.ReplicationProvider.Equals(Constants.HyperVReplicaAzure))
-                || (this.VmmToVmm.IsPresent 
+                || (this.VmmToVmm.IsPresent
                 && !(this.ReplicationProtectedItem.ReplicationProvider.Equals(Constants.HyperVReplica2012)
                 || this.ReplicationProtectedItem.ReplicationProvider.Equals(Constants.HyperVReplica2012R2))))
             {
                 throw new PSInvalidOperationException(Resources.InvalidParameterSet);
             }
         }
 
+        /// <summary>
+        ///     validate parameter vaules for recovery plan.
+        /// </summary>
         private void validateRPSwitchParam(RecoveryPlan rp, string replicationProvider)
```

Comment:
Is there any way to move this validation into parameter validation instead of run time validation?

Reply:
ReplicationProvider in internal property of RecoveryPlan.

Do we have any example of adding validation of input paramset with internal property of passed object?

Reply:
Looking again this does not look like it would be possible to do.